client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -83,98 +144,72 @@ export class SiteVerticalsSuggestionSearch extends Component {
 				verticalSlug: value,
 			}
 		);
+	};
 
-	onSiteTopicChange = ( value, isNavigating ) => {
-		const hasValue = !! value;
-		const valueLength = value.length || 0;
-		const valueLengthShouldTriggerSearch = valueLength >= this.props.charsToTriggerSearch;
-		const result = this.searchForVerticalMatches( value );
-
-		// TODO:
-		// Where to put the railcar code will be reconsidered.
+	/***
+	 * Callback to be passed to consuming component when the search field is updated.
+	 *
+	 * @param {String} value The new search value
+	 * @param {Bool} isSuggestionSelected Whether the user has selected a suggestion
+	 */
+	onSiteTopicChange = ( value, isSuggestionSelected ) => {
+		// TODO: Where to put the railcar code will be reconsidered.
 		if (
-			hasValue &&
-			valueLengthShouldTriggerSearch &&
-			// Don't trigger a search if there's already an exact, non-user-defined match from the API
-			! result
+			!! value &&
+			value !== this.props.searchValue &&
+			size( value ) >= this.props.charsToTriggerSearch
 		) {
 			this.setState( { railcar: this.getNewRailcar() } );
```

Comment:
@adamziel 

Since we're not using this condition block, what would be the conditions for railcar now?

Reply:
My two cents for `value && value !== this.props.searchValue` since it seems that we can deprecate `chartsToTriggerSearch` . If this sounds good, I think we can also remove that TODO comment. I wrote that because I thought it wasn't the best place, but it now looks about right after your work here :)